From b353cdb59fd107006a9cbaf03f418f61bd55e8d2 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Tue, 18 Mar 2025 16:19:43 +0530
Subject: [PATCH 2/2] HACK: hardcode SoC name as TDA54

Currently, SoC name is not available from sysfs userspace in tda54-vdk
platform. Therefore, hardcode the SoC name to pickup TDA54 map and launch
IPC on this SoC.

Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 examples/rpmsg_char_simple.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/examples/rpmsg_char_simple.c b/examples/rpmsg_char_simple.c
index 84f2708..a363681 100644
--- a/examples/rpmsg_char_simple.c
+++ b/examples/rpmsg_char_simple.c
@@ -210,8 +210,8 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
-	/* Use auto-detection for SoC */
-	ret = rpmsg_char_init(NULL);
+	/* Hard-code SoC name as SoC info is not available in sysfs */
+	ret = rpmsg_char_init("TDA54");
 	if (ret) {
 		printf("rpmsg_char_init failed, ret = %d\n", ret);
 		return ret;
-- 
2.34.1

